c++/include/orc/Reader.hh

```diff
@@ -565,6 +591,15 @@ namespace orc {
      */
     virtual void seekToRow(uint64_t rowNumber) = 0;
 
+    /**
+     * Get an ArrayReadIntent for a given typeId.
+     * @param typeId the type id to look up.
+     * @return an ArrayReadIntent that was specified for given typeId through
+     *    RowReaderOptions::includeTypesWithIntents. If no ArrayReadIntent was
+     *    specified for typeId, return ArrayReadIntent_ALL.
+     */
+    virtual ArrayReadIntent getReadIntent(uint64_t typeId) const = 0;
```

Comment:
My plan is to return `ArrayReadIntent_ALL` if typeId is not selected or selected without any specific read intent (in case `RowReaderOptions::includeTypes()` is being used). I understand the naming is a bit confusing. Will think more about this.

Reply:
Deleted in latest commit.